audbcards/core/dataset.py

```diff
@@ -390,79 +462,131 @@ def create_datacard_page(dataset: Dataset):
-                if has_mappings:
-                    if not isinstance(scheme.labels, (str, dict)):
-                        mappings = ''
-                    else:
-                        labels = scheme._labels_to_dict()
-                        # Mappings can contain a single mapping
-                        # or a deeper nestings.
-                        # In the first case we just present ✓,
-                        # in the second case the keys of the nested dict.
-                        # {'f': 'female', 'm': 'male'}
-                        # or
-                        # {'s1': {'gender': 'male', 'age': 21}}
-                        mappings = list(labels.values())
-                        if isinstance(mappings[0], dict):
-                            # e.g. {'s1': {'gender': 'male', 'age': 21}}
-                            mappings = sorted(list(mappings[0].keys()))
-                            mappings = f'{", ".join(mappings)}'
-                        else:
-                            # e.g. {'f': 'female', 'm': 'male'}
-                            mappings = '✓'
-                        fp.write(f', "{mappings}"')
-                fp.write('\n')
-
-
-def create_datasets_page(datasets: typing.Sequence):
-    r"""Create overview page of datasets."""
-    # Create CSV file with overview of datasets
+
+        columns = ['ID', 'Dtype']
+        header_line = '   :header: ID,Dtype'
+        if has_minimums:
+            header_line += ',Min'
+            columns.append('Min')
+        if has_maximums:
+            header_line += ',Max'
+            columns.append('Max')
+        if has_labels:
+            header_line += ',Labels'
+            columns.append('Labels')
+        if has_mappings:
+            header_line += ',Mappings'
+            columns.append('Mappings')
+
+        scheme_info['header_line'] = header_line
+        scheme_info['has_minimums'] = has_minimums
+        scheme_info['has_maximums'] = has_maximums
+        scheme_info['has_labels'] = has_labels
+        scheme_info['has_mappings'] = has_mappings
+
+        scheme_info['columns'] = columns
+        return scheme_info
+
+    @property
+    def dataset_schemes(self) -> list:
+
+        db = self.header
+        dataset_schemes = []
+        for scheme_id in db.schemes:
+            dataset_scheme = self._scheme_to_list(scheme_id)
+            dataset_schemes.append(dataset_scheme)
+
+        cols = self.scheme_info['columns']
+        data = pd.DataFrame.from_dict(dataset_schemes)[cols]
+        filter = data.applymap(lambda d: d == [])
+        data.mask(filter, other='', inplace=True)
+        scheme_data = np.array(data).tolist()
+        return scheme_data
+
+
+class Datacard(object):
+
+    def __init__(self, dataset: Dataset):
+
+        self._dataset = dataset
+
+    @property
+    def props(self) -> dict:
+        """slot to hold superclass properties for rendering."""
+        props = self._dataset.properties()
+        return props
+
```

Comment:
When my other comment is applied this is no longer needed:

```suggestion
```

Reply:
Your suggestion is to replace the `Dataset.properties` method A

```python
    """Additions to class """
    def properties(self):
        """Get list of properties of the object."""

        class_items = self.__class__.__dict__.items()
        props = dict((k, getattr(self, k))
                     for k, v in class_items
                     if isinstance(v, property))
```

with B

```python
properties = [
    p for p in dir(audbcards.Dataset)
    if isinstance(getattr(audbcards.Dataset, p), property)
]
```

Afaics A returns a `dict` that contains the values, whereas B contains consists of the list of property names only. 
In other words `list(A.keys())`  should be identical to `properties` in B.

Note that starting the rendering process requires a dictionary containing the values:

```python
content = template.render(self.props)
```

So to make B work, one would have to get the property values from the object instance in a second step. 
So `getattr` as in A will be needed eventually.

Reply:
No, I would no longer replace `Dataset.properties()`.
The request here is to remove the `Datacard.props` property as I would not include it in the public API, and instead directly call `self._dataset.properties()` instead of `self.props` inside `Datacard`.